src/arrayMethodJoin.js

```diff
@@ -4,8 +4,22 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joined = '';
+
+    for (let i = 0; i < this.length - 1; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        joined += `${separator}`;
+      } else {
+        joined += `${this[i]}${separator}`;
+      }
+    }
```

Comment:
you can create a separate variable for the value (`${separator}`), to not repeat yourself